Update to build with rust nightly

LukasKalbertodt opened the pull request:
I replaced all uses of `std::old_io` with functions of `std::io`. Now it compiles again with my rust nightly version ( see #47 ).

I'm really not sure what I did here and if it's useful for you at all...

LukasKalbertodt commented:
Testing causes an internal compiler error. Seems to be an well known issue of the new nightly. Related: https://github.com/rust-lang/rust/issues/23068

reem commented:
@LukasKalbertodt thanks for this fix! Do you mind if I hold off on merging until travis is fixed?

LukasKalbertodt commented:
Uhm... thats new. The new openssl version fails to compile. I don't mind waiting until they fixed the ICE ... should I revert my last "workaround"-commit then?

reem commented:
There's nothing really to be done until travis gets fixed, so I'd just wait until we know more.

s-panferov commented:
@reem I think that it must work already. Can you re-run travis build?

reem commented:
Everything is building! @LukasKalbertodt can you try removing that last workaround commit, I don't think it's needed anymore.

s-panferov commented:
:tada:

s-panferov commented:
@LukasKalbertodt I think now it will be better to remove these commits with:

~~~
git reset --hard HEAD~2 && git push LukasKalbertodt refactor/new-io
~~~

LukasKalbertodt commented:
@s-panferov I'm scared of `--hard` and `-f`, but the commits are gone now ;-)
Travis need a rebuild...

fsommar commented:
Is it possible to force travis to rebuild this PR so that it can be merged into master?

s-panferov commented:
@reem could you please merge this to unblock rustless

reem commented:
Retrying travis now

LukasKalbertodt commented:
I don't get it. I can't reproduce that build error locally. I updated my Rust and Cargo, deleted `target` and `Cargo.lock` as well as everything in `~/.cargo`. Works fine for me (I have `matches v0.1.2` too) ...

reem commented:
The travis version might be cached a bit too far or something. I'm just gonna merge this anyway.
